### DISCUSSION TITLE
Angular application

### DISCUSSION THREAD

**Discussion by robertwakeling** [MathKeyboardEngine/MathKeyboardEngine#8] [Q&A] [answered]
Hey I think your MathKeyboard is amazing! 
I have the files in my node_modules folder but I'm having difficulty getting it to display correctly in my Angular application. I've tried pasting all of the code into an html file and it comes out like this:
![Katex](https://github.com/MathKeyboardEngine/MathKeyboardEngine/assets/1598750/9259f877-e83c-4cd5-9dd8-84034c4a5597)

Could anybody please advise how I would get this working?
Yours in hope!
Rob

**Comment by SymboLinker** [comment, marked as answer]
Hi Robert,

My apologies for not providing better guidance - I'm not familiar with Angular and I am packing for holiday. I'll only give broad guidelines.

Take smaller steps and test each step.

Step 1
Try to `console.log` a LaTeX result that MathKeyboardEngine has generated. This can be as simple as the result of:
```
import * as mke from 'mathkeyboardengine';

const k = new mke.KeyboardMemory();
const config = new mke.LatexConfiguration();
var output = mke.getEditModeLatex(k, config);
```

Step 2
Try to display the result from the previous step on any page, so that you see `\blacksquare` in your browser.

Step 3
Try to display this as rendered LaTeX. For Angular, you may need to use a package like:
https://www.npmjs.com/package/ng-katex

Step 4
Make sure you understand Angular so well, that you know how to handle button clicks and re-rendering the content. After that you can start copy-pasting larger chuncks of code from the MathKeyboardEngine examples.

Keep trying... you'll get it working! (And feel free to ask more detailed questions here, so that we can help you further if you're stuck.)

Kind regards,
SymboLinker

**Reply by robertwakeling** [reply]
Thank you so much; I managed to get it working now :)

**Reply by SymboLinker** [reply]
Well done! I'm glad I could help.

Maybe somewhere in the future: feel free to promote your (web)app at the [Show and tell](https://github.com/MathKeyboardEngine/MathKeyboardEngine/discussions/categories/show-and-tell) section.
